instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -564,11 +564,11 @@ def _instrument(self, **kwargs):
         tracer_provider = kwargs.get("tracer_provider")
         _InstrumentedClient._tracer_provider = tracer_provider
         _InstrumentedAsyncClient._tracer_provider = tracer_provider
-        httpx.Client = _InstrumentedClient
+        httpx.Client = httpx._api.Client = _InstrumentedClient
```

Comment:
Wait, this is using a private attribute and subject to unexpected breaking changes. Is there another way to do this patching that does not involve setting a private attribute?

Reply:
Definitely the easiest way by far. :) Other than instrumenting methods separately, I don't easily how to make it work otherwise.

There's an [issue open in httpx repo](https://github.com/encode/httpx/issues/3176) documenting 3rd party usage of private API. If this is merged, I can leave a note there about this instrumentation.

Reply:
@ocelotl, would you reconsider the approval and merge this?

Reply:
Ok, I'll just add a comment here explaining why we are using a private attribute :v: